src/users/users.controller.ts

```diff
@@ -72,11 +75,56 @@ export class UsersController {
     return this.usersService.update(id, updateUserDto);
   }
 
+  @Put('username/:username')
+  @Roles([UserRoleEnum.ADMIN, UserRoleEnum.USER])
+  updateByUsername(
+    @Param('username') username: string,
+    @Body() updateUserDto: UpdateUserDto,
+  ): Promise<UserModel> {
+    const message = `UsersController.updateByUsername() username=${username} updateUserDto=${JSON.stringify(
+      updateUserDto,
+    )}`;
+    this.logger.log(message);
+    return this.usersService.updateByUsername(username, updateUserDto);
+  }
```

Comment:
let's use [@Query](https://docs.nestjs.com/controllers#request-object) in `PUT /users/:id`. by using `@Query`, we can avoid creating a new route. Check [this post](https://www.reddit.com/r/node/comments/bwftpk/how_to_use_query_parameters_for_searching_a/) for your reference.